### ISSUE TITLE
Add feature to distinguish different metric points from different places on the same metric graph

### ISSUE THREAD

**Issue by lnxslck** [CachetHQ/Cachet#2091]
Before submitting your issue, please make sure that you've checked the checkboxes below.

- [x ] I am running the [latest release](https://github.com/CachetHQ/Cachet/releases/latest) version of Cachet.
- [ x] I am running at least PHP 5.5.9. *You can check this by running `php -v`.*
- [ ] I have ran `rm -rf bootstrap/cache/*`.

### Expected behaviour

Let's say we have one metric graph, that receives metric points from 2 different servers. 
It would be a nice feature if we could see that those metric points come from different places, different colours or labels, or something like that.

### Actual behaviour

All metric points show in the metric graph, but we don't have a way to distinguish from were they come.

### Steps to reproduce

*If your issue requires any specific steps to reproduce, please outline them here.*

1. Create a metric graph
2. Add metric points from different servers

**Comment by ConnorVG** [comment]
This would require a slightly radical change to the way the metric points work, though we could do something like _metric sources_.

idk...

**Comment by lnxslck** [comment]
@ConnorVG Maybe just a different color for each "different" metric point? 

Let's say you have a component group with 3 components, so each component could have a colour ?

**Comment by ConnorVG** [comment]
:rainbow:

**Comment by classicyuppie** [comment]
I have to agree with @lnxslck - [this is something I'm needing for my own project](https://github.com/CachetHQ/Cachet/issues/2091#issuecomment-244009066). We have several metric points on the same graph for multiple components which doesn't make it easy to compare for multiple components on the _same_ graph.

**Comment by jbrooksuk** [comment]
@classicyuppie makes sense. Metrics really need a complete rewrite to fix a few of issues with them. We could also do with a better way to graph them too.
